src/include/distributed/multi_physical_planner.h

```diff
@@ -169,6 +169,9 @@ typedef struct Task
 	uint64 shardId;                /* only applies to shard fetch tasks */
 	TaskExecution *taskExecution;  /* used by task tracker executor */
 	bool upsertQuery;              /* only applies to modify tasks */
+
+	bool insertSelectQuery;
```

Comment:
> What if we add an explicit routerExecutable flag to the job?

I think that makes sense and works well. Do you think is it better to have it in this PR, or in the other PR?

> or at least writes with a co-located subquery

We allow subqueries to include reference tables (i.e., join with colocated table). Does that break locking proposed here? (which I don't think so, but, good to raise for your attention)